app/algorithms/normal.ts

```diff
@@ -0,0 +1,121 @@
+import isEmpty = require('lodash/isEmpty');
+
+// This is a TypeScript file in a JavaScript project so it must be complied
+// If you make changes to this file it must be recomplied using `tsc` in
+// order for the changes to be reflected in the rest of the program.
+//
+// app/models/election .elect() is the only file that uses this function
+// and importes it from normal.js, which is the compiled result of this file.
+//
+export type Normal = {
+  result: NormalResult;
+  thr: number;
+  voteCount: number;
+  blankVoteCount: number;
+  useStrict: boolean;
+};
+
+type Alternative = {
+  _id: string;
+  description: string;
+  election: string;
+};
+
+type Vote = {
+  _id: string;
+  priorities: Alternative[];
+};
+
+enum Status {
+  resolved = 'RESOLVED',
+  unresolved = 'UNRESOLVED',
+}
```

Comment:
I think this should be a global type shared between the two algorithms. This also goes for types like `Alternative` and any other shared types. Put them all in an types.d.ts pls.